SpatialConnect/SCExchangeAuthMethod.m

```diff
@@ -44,24 +43,57 @@ - (BOOL)authFromCache {
 
 - (BOOL)authenticate:(NSString *)u password:(NSString *)p {
   username = u;
-  NSURL *url =
-  [NSURL URLWithString:[NSString stringWithFormat:@"%@/o/token",
-                        serverUrl]];
-  NSDictionary *authDict = @{ @"grant_type": @"password", @"username" : u, @"password" : p };
-  NSString *oauthCreds = [NSString stringWithFormat:@"%@:%@", clientId, clientSecret];
+  NSURL *url = [NSURL
+      URLWithString:[NSString stringWithFormat:@"%@/o/token/", serverUrl]];
+  NSString *auth = [NSString
+      stringWithFormat:@"grant_type=password&username=%@&password=%@", u, p];
+  NSData *authBody = [auth dataUsingEncoding:NSUTF8StringEncoding];
+
+  NSString *oauthCreds = [NSString stringWithFormat:@"%@:", clientId];
   NSData *nsdata = [oauthCreds dataUsingEncoding:NSUTF8StringEncoding];
   NSString *base64Encoded = [nsdata base64EncodedStringWithOptions:0];
-  NSString *auth = [NSString stringWithFormat:@"Basic %@", base64Encoded];
-  NSDictionary *res =
-  [SCHttpUtils postDictRequestAsDictBLOCKING:url body:authDict auth:auth];
-  if (res && (jwt = res[@"result"][@"token"])) {
+  NSString *authHeader = [NSString stringWithFormat:@"Basic %@", base64Encoded];
+  NSDictionary *res = [SCHttpUtils postDataRequestAsDictBLOCKING:url
+                                                            body:authBody
+                                                            auth:authHeader
+                                                     contentType:XML];
+
+  if (res && (jwt = res[@"access_token"])) {
     [keychainItem setObject:p forKey:(__bridge id)kSecValueData];
     [keychainItem setObject:username forKey:(__bridge id)kSecAttrAccount];
+    [keychainItem setObject:jwt forKey:(__bridge id)kSecAttrGeneric];
```

Comment:
@tetriscode @frankrowe 
Is this the best way to store the token in the keychain? or is there a better way storing the token for later use?

Reply:
I guess I could store it in `SCCache`